### PR TITLE
Support CreatedConnection and Backfill support

### PR DESCRIPTION
**Pull request by jdneumeyer77**
Includes some refactoring to more closely resemble the Google Ads connector and make removing the old subscription event easier to remove in the future.

### CODE REVIEW

File: backend/project/LibraryDependencies.scala [hunk 1]
```diff
@@ -2,7 +2,7 @@ import sbt._
 
 object LibraryDependencies {
   object NarrativeBackend {
-    val version = "20.0.28"
+    val version = "20.11.5"
```
Comment (mbabic) [inline backend/project/LibraryDependencies.scala:5]: Thank you. :)

File: backend/stores/src/main/scala/io/narrative/connectors/facebook/domain/Job.scala [hunk 2]
```diff
@@ -4,10 +4,12 @@ import cats.{Eq, Show}
 import doobie.Meta
 import io.circe.{Codec, Decoder, Encoder}
 import io.circe.generic.extras.semiauto.deriveConfiguredCodec
+import io.narrative.connectors.model.SnapshotId
 
 final case class Job(
     eventRevision: Revision,
-    file: FileName
+    file: FileName,
+    snapshotId: Option[SnapshotId] // TODO: not optional after Subscription removal.
```
Comment (mbabic) [inline backend/stores/src/main/scala/io/narrative/connectors/facebook/domain/Job.scala:12]: As a follow up we can drop support for legacy subscriptions at this point. You can cut a ticket to do that and throw it in the tech debt backlog, if you're looking for work one day feel free to pick it up.

File: backend/worker/src/main/scala/io/narrative/connectors/facebook/CommandProcessor.scala [hunk 3]
```diff
@@ -34,8 +39,9 @@ class CommandProcessor(
       toConnectionIO: IO ~> ConnectionIO
   ): ConnectionIO[CommandProcessor.Result] =
     for {
-      _ <- doobie.free.connection.delay(logger.info(s"processing ${event}"))
+      _ <- logger.info(s"processing ${event}")
```
Comment (mbabic) [inline backend/worker/src/main/scala/io/narrative/connectors/facebook/CommandProcessor.scala:42]: +1 for this refactor.

File: backend/worker/src/main/scala/io/narrative/connectors/facebook/DeliveryProcessor.scala [hunk 4]
```diff
@@ -3,25 +3,29 @@ package io.narrative.connectors.facebook
...
 import io.narrative.connectors.facebook.services.{FacebookClient, FacebookToken, TokenEncryptionService}
 import io.narrative.connectors.facebook.stores.CommandStore.StatusUpdate.FileUpdate
 import io.narrative.connectors.facebook.stores.{CommandStore, ProfileStore, SettingsStore}
 import io.narrative.connectors.spark.ParquetTransformer
+import org.typelevel.log4cats.{LoggerFactory, SelfAwareStructuredLogger}
+import io.narrative.connectors.facebook.domain.ConnectionId
+import io.narrative.connectors.facebook.domain.Audience
+import io.narrative.connectors.model.DatasetId
 
+// TODO: handle connection creation events?
```
Comment (mbabic) [inline backend/worker/src/main/scala/io/narrative/connectors/facebook/DeliveryProcessor.scala:28]: Nit: you came back and handled this TODO. :)

File: backend/stores/src/main/scala/io/narrative/connectors/facebook/domain/Job.scala (hunk 2 above)
Comment (jdneumeyer77) [reply backend/stores/src/main/scala/io/narrative/connectors/facebook/domain/Job.scala:12]: Will do.

**Pull request merged**